sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -301,53 +293,64 @@ export class EducationProgramOfferingController {
   @Groups(UserGroups.AESTUser)
-  @Get("institution/:institutionId/programs")
-  async getPaginatedProgramsForInstitution(
-    @Param("institutionId") institutionId: number,
-    @Query("pageSize") pageSize: number,
-    @Query("page") page: number,
-    @Query("sortColumn") sortColumn: string,
-    @Query("sortOrder") sortOrder: SortDBOrder,
-    @Query("searchProgramName") searchProgramName: string,
-  ): Promise<ProgramsOfferingSummaryPaginated> {
-    const paginatedProgramOfferingSummaryResult =
-      await this.programOfferingService.getPaginatedProgramsForInstitution(
-        institutionId,
-        pageSize,
-        page,
-        sortColumn,
-        sortOrder,
-        searchProgramName,
-      );
-    const paginatedProgramOfferingSummary =
-      paginatedProgramOfferingSummaryResult.programsSummary.map(
-        (programOfferingSummary) => ({
-          programId: programOfferingSummary.programId,
-          programName: programOfferingSummary.programName,
-          submittedDate: programOfferingSummary.submittedDate,
-          formattedSubmittedDate: getDateOnlyFormat(
-            programOfferingSummary.submittedDate,
-          ),
-          locationName: programOfferingSummary.locationName,
-          programStatus: programOfferingSummary.programStatus,
-          offeringsCount: programOfferingSummary.offeringsCount,
-        }),
+  @Get("location/:locationId/education-program/:programId/aest")
+  async getOfferingSummary(
+    @Param("locationId") locationId: number,
+    @Param("programId") programId: number,
+    @Query("searchName") searchName: string,
+    @Query("sortField") sortField: string,
+    @Query("sortOrder") sortOrder = FieldSortOrder.ASC,
+    @Query("page") page = DEFAULT_PAGE_NUMBER,
+    @Query("pageLimit") pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<PaginatedOffering> {
+    //To retrieve Education program offering corresponding to ProgramId and LocationId
+    // [OfferingTypes.applicationSpecific] offerings are created during PIR, if required
+    return this.programOfferingService.getAllEducationProgramOffering(
+      locationId,
+      programId,
+      {
+        searchName: searchName,
+        sortField: sortField,
+        sortOrder: sortOrder,
+        page: page,
+        pageLimit: pageLimit,
+      },
+      [OfferingTypes.public],
+    );
+  }
+
+  /**
+   * Offering details for ministry users
+   * @param offeringId offering id
+   * @returns Offering details
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Get(":offeringId/aest")
+  async getProgramOfferingForAEST(
+    @Param("offeringId") offeringId: number,
+  ): Promise<ProgramOfferingDto> {
+    //To retrieve Education program offering corresponding to ProgramId and LocationId
+    const offering = await this.programOfferingService.getOfferingById(
+      offeringId,
+    );
+    if (!offering) {
+      throw new UnprocessableEntityException(
```

Comment:
I believe that it should be 404 because the id, that is part of the URL does not exists, right?